server/sonar-server/src/main/java/org/sonar/server/language/ws/ListAction.java

```diff
@@ -80,7 +80,7 @@ void define(WebService.NewController controller) {
   }
 
   private Collection<Language> listMatchingLanguages(@Nullable String query, int pageSize) {
-    Pattern pattern = Pattern.compile(query == null ? MATCH_ALL : MATCH_ALL + query + MATCH_ALL, Pattern.CASE_INSENSITIVE);
+    Pattern pattern = Pattern.compile(query == null ? MATCH_ALL : MATCH_ALL + Pattern.quote(query) + MATCH_ALL, Pattern.CASE_INSENSITIVE);
```

Comment:
I guess this is old debt, by it's kind of waste (to not say "stupid") to use a regexp pattern when the query is null
one could just not create a pattern and directly apply paging to `languages.all`
it just takes making the paging code a function...